src/LibraryUtilities.ts

```diff
@@ -250,6 +286,34 @@ export function constructLibraryItem(
 }
 
 /**
+ * Compare two arrays of strings and check if the strings in the first array match
+ * those in the second array.
+ * 
+ * @param {string[]} parts 
+ * The array of strings to be checked.
+ * 
+ * @param {string[]} partsToInclude 
+ * The array of strings that should be present in 'parts'.
+ * 
+ * @returns
+ * Returns 0 if the two inputs match completely.
+ * Returns how many items 'parts' has more than 'partsToInclude', provided that
+ * 'parts' contains all items in 'partsToInclude'.
+ * Returns undefined if the two inputs do not match.
+ * 
+ */
+function leftoverPart(parts: string[], partsToInclude: string[]): number | undefined {
+    let leftover = parts.length;
+    for (let i = 0; i < partsToInclude.length; i++) {
```

Comment:
I think the `i` is necessary since the index needs to be used to compare both `parts` and `partsToInclude`.